src/components/ConversationSettings/DetailsSettings.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+	<Fragment>
+		<NcTextField :value.sync="conversationNameTextField"
+			:label="t('spreed', 'Conversation Name')"
+			:disabled="!isEditingDetails"
+			:label-visible="true" />
+		<NcTextField :value.sync="descriptionTextField"
+			:label="t('spreed', 'Conversation description')"
+			:placeholder="t('spreed', 'Enter a description for this conversation')"
+			:disabled="!isEditingDetails"
+			:label-visible="true" />
+		<div class="details__buttons">
+			<NcButton v-if="!isEditingDetails" type="secondary" @click="isEditingDetails = true">
+				{{ t('spreed', 'Edit details') }}
+			</NcButton>
+			<template v-else-if="isEditingDetails">
+				<NcButton type="tertiary" @click="cancelEditing">
+					{{ t('spreed', 'Cancel') }}
+				</NcButton>
+				<NcButton type="secondary" @click="handleUpdateDetails">
+					{{ t('spreed', 'Save') }}
+				</NcButton>
+			</template>
+		</div>
+	</Fragment>
+</template>
+
+<script>
+import { showError } from '@nextcloud/dialogs'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import { Fragment } from 'vue-frag'
+
+export default {
+	name: 'DetailsSettings',
+
+	components: {
+		NcTextField,
+		NcButton,
+		Fragment,
+	},
+
+	props: {
+		/**
+		 * The conversation object
+		 */
+		conversation: {
+			type: Object,
+			required: true,
+		},
+
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			isEditingDetails: false,
+			conversationNameTextField: '',
+			descriptionTextField: '',
+		}
+	},
+
+	watch: {
+		// Update details everytime the conversation object changes
```

Comment:
that got me confused, but the return when editing is important, maybe add that to the comment